platform/src/main/java/org/stellar/anchor/platform/payment/observer/stellar/PaymentObservingAccountsManager.java

```diff
@@ -0,0 +1,157 @@
+  Map<String, ObservingAccount> allAccounts;
+  private final PaymentObservingAccountStore store;
+
+  public PaymentObservingAccountsManager(PaymentObservingAccountStore store) {
+    this.store = store;
+    allAccounts = new ConcurrentHashMap<>();
+  }
+
+  @PostConstruct
+  public void initialize() throws ValueValidationException {
+    List<PaymentObservingAccount> accounts = store.list();
+    for (PaymentObservingAccount account : accounts) {
+      ObservingAccount oa =
+          new ObservingAccount(account.getAccount(), account.getLastObserved(), true);
+      upsert(oa);
+    }
+
+    // Start the eviction task
+    ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
+    scheduler.scheduleAtFixedRate(
+        this::evictAndPersist, 60, getEvictPeriod().getSeconds(), TimeUnit.SECONDS);
+  }
+
+  public void evictAndPersist() {
+    Log.info("Evicting old accounts...");
+    this.evict(getEvictMaxAge());
+    Log.info("Persisting accounts...");
+    for (ObservingAccount account : this.getAccounts()) {
+      store.upsert(account.account, account.lastObserved);
+    }
+  }
+
+  /**
+   * Adds an account to be observed. If the account is being observed, it will be updated.
+   *
+   * @param account The account being observed.
+   * @param expiring true if the account can be expired. false, otherwise.
+   */
+  public void upsert(String account, Boolean expiring) throws ValueValidationException {
+    if (account != null && expiring != null) {
+      upsert(new ObservingAccount(account, Instant.now(), expiring));
+    }
+  }
+
+  /**
+   * Add an account to be observed. If the account is being observed, it will be updated.
+   *
+   * @param observingAccount The account being observed.
+   */
+  public void upsert(ObservingAccount observingAccount) throws ValueValidationException {
+    if (observingAccount != null) {
+      ObservingAccount existingAccount = allAccounts.get(observingAccount.account);
+      if (existingAccount == null) {
+        allAccounts.put(observingAccount.account, observingAccount);
+        // update the database
+        store.upsert(observingAccount.account, observingAccount.lastObserved);
+      } else {
+        if (!observingAccount.expiring.equals(existingAccount.expiring))
+          throw new ValueValidationException(
+              String.format(
+                  "The expiring flag cannot be modified. Account=[%s]", observingAccount.account));
+        existingAccount.account = observingAccount.account;
+        existingAccount.lastObserved = observingAccount.lastObserved;
+      }
+    }
+  }
+
+  /**
+   * Remove the account from the observed account list.
+   *
+   * @param account The account being removed.
+   */
+  public void remove(String account) {
+    if (account != null) {
+      ObservingAccount targetAccount = allAccounts.get(account);
+      if (targetAccount != null) {
+        allAccounts.remove(account);
+      }
+    }
+  }
```

Comment:
❓  Does it make sense to remove the account from the store too?

Reply:
There is no rea; use case for now. It does not hurt to keep it here but it has no immediate use either. Do you prefer to remove?

Reply:
I think we should be consistent with the `evict` method where we're removing it from the store.

https://github.com/stellar/java-stellar-anchor-sdk/blob/eb7fa5b76d213c2c430f79b563c19c12b69139b9/platform/src/main/java/org/stellar/anchor/platform/payment/observer/stellar/PaymentObservingAccountsManager.java#L126-L141

Reply:
Good point. I have removed the `remove()` function.